backend/useragent/user_agent.go

```diff
@@ -6,7 +6,7 @@ import (
 )
 
 var (
-	userAgentRegex   = regexp.MustCompile(`^Grafana/([0-9]+\.[0-9]+\.[0-9]+(?:-[a-zA-Z0-9]+)?) \(([a-zA-Z0-9]+); ([a-zA-Z0-9]+)\)$`)
+	userAgentRegex   = regexp.MustCompile(`^Grafana/([0-9]+\.[0-9]+\.[0-9]+(?:\+[a-zA-Z0-9-]+)?(?:-[a-zA-Z0-9]+)?) \(([a-zA-Z0-9]+); ([a-zA-Z0-9]+)\)$`)
```

Comment:
maybe we can simplify this a bit and capture everything after the three version numbers?

```suggestion
	userAgentRegex   = regexp.MustCompile(`^Grafana/([0-9]+\.[0-9]+\.[0-9]+(?:[^\s]+)?) \(([a-zA-Z0-9]+); ([a-zA-Z0-9]+)\)$`)
```

Reply:
I guess this will mixup with the patch version number, maybe adding `[-\+]` in front like this?

```suggestion
	userAgentRegex   = regexp.MustCompile(`^Grafana/([0-9]+\.[0-9]+\.[0-9]+(?:[-\+][^\s]+)?) \(([a-zA-Z0-9]+); ([a-zA-Z0-9]+)\)$`)
```

Reply:
how would it mixup with the patch version number? `[0-9]+` will match all numbers of the patch version and the rest would be the pre-release. In any case, we don't capture version numbers independently so we could go with the simplest one.

Reply:
Just chiming in as I originally wrote this. The idea was to be super simple at the time, but obviously it's a bit _too_ simple 😅 This is fine I guess as long as you can't validate an invalid semver with this.

The alternative I would throw out there is to just add the semver provided from [semver.org](https://semver.org/) - so we'll likely never have to touch this again 🤷

Reply:
Maybe we should keep the regexp here as simple as possible, and add something more powerful, like the semver regexp to the build process?

I have no idea how many times this code is invoked if it works, but the combination of the `11.2.2+security-01` version and the broken regexp produce one warning every ~10s on my installation...

Reply:
Thinking on it some more, we don't really want to couple Grafana's version naming strategy here with logic inside the SDK, because at the end of the day this is purely informational. I think this is fine for now to get the logs clear, but might revisit this to tidy it up a bit